__tests__/index.test.js

```diff
@@ -124,97 +124,129 @@ describe("GET /api/articles", () => {
-              expect(body.article_id).toBe(1);
-              expect(body.body).toBe("This is a test comment");
-          });
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ username: "butter_bridge", body: "This is a test comment" })
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment_id).toBe(19);
+        expect(body.author).toBe("butter_bridge");
+        expect(body.article_id).toBe(1);
+        expect(body.body).toBe("This is a test comment");
+      });
+  });
+
+  test("status 201: responds with the posted comment object ignoring any extra properties", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({
+        username: "butter_bridge",
+        body: "This is a test comment",
+        extra: "extra",
+      })
+      .expect(201)
+      .then(({ body }) => {
+        expect(body.comment_id).toBe(19);
+        expect(body.author).toBe("butter_bridge");
+        expect(body.article_id).toBe(1);
+        expect(body.body).toBe("This is a test comment");
+      });
   });
 
   //Error handling
   test("status 404: responds with error message when the given article_id does not exist", () => {
-      return request(app)
-          .post("/api/articles/1000/comments")
-          .send({username: "butter_bridge", body: "This is a test comment"})
-          .expect(404)
-          .then(({body}) => {
-              expect(body.msg).toBe("Article not found");
-          });
+    return request(app)
+      .post("/api/articles/1000/comments")
+      .send({ username: "butter_bridge", body: "This is a test comment" })
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Article not found");
+      });
   });
 
   test("status 400: responds with error message when given an invalid article_id", () => {
-      return request(app)
-          .post("/api/articles/invalid/comments")
-          .send({username: "butter_bridge", body: "This is a test comment"})
-          .expect(400)
-          .then(({body}) => {
-              expect(body.msg).toBe("Bad request");
-          });
+    return request(app)
+      .post("/api/articles/invalid/comments")
+      .send({ username: "butter_bridge", body: "This is a test comment" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
   });
 
-});
+  test("status 400: responds with error message when no body is given", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ username: "butter_bridge" })
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Bad request");
+      });
+  });
 
+  test("status 400: responds with error message when given non-existent username", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({ username: "invalid", body: "This is a test comment" })
+      .expect(400)
```

Comment:
This is technically just non-existent, but it _could_ exist, so therefore could be a 404 instead of a 400